### ISSUE TITLE
[BUG] nightly spark303 deploy pulling spark301 aggregator

### ISSUE THREAD

**Issue by pxLi** [NVIDIA/spark-rapids#4356]
**Describe the bug**
This cannot be reproduced locally, only in nightly pipeline

`mvn -B -Dbuildver=303 deploy -pl '!tools,!dist' -s jenkins/settings.xml -P mirror-apache-to-urm -Dmaven.repo.local=/home/jenkins/agent/workspace/jenkins-rapids_nightly-dev-github-410/.m2 -Dcuda.version=cuda11 -DskipTests`
```
09:52:52  [INFO] ----------------< com.nvidia:rapids-4-spark-tests_2.12 >----------------
09:52:52  [INFO] Building RAPIDS Accelerator for Apache Spark Tests 22.02.0-SNAPSHOT [9/11]
09:52:52  [INFO] --------------------------------[ jar ]---------------------------------
09:52:52  [INFO] Downloading from snapshots: https://urm.nvidia.com/artifactory/sw-spark-maven/com/nvidia/rapids-4-spark-aggregator_2.12/22.02.0-SNAPSHOT/rapids-4-spark-aggregator_2.12-22.02.0-20211211.165958-291-spark301.jar
09:52:52  [INFO] Downloaded from snapshots: https://urm.nvidia.com/artifactory/sw-spark-maven/com/nvidia/rapids-4-spark-aggregator_2.12/22.02.0-SNAPSHOT/rapids-4-spark-aggregator_2.12-22.02.0-20211211.165958-291-spark301.jar (11 MB at 42 MB/s)
```

**Comment by pxLi** [comment]
intermittently it could be reproduced locally, could be related to https://github.com/NVIDIA/spark-rapids/pull/4265/files

I would guess some commands in maven plugin do work consistently in different ENVs

**Comment by jlowe** [comment]
@tgravescs and I believe this is triggered by the dist jar depending upon the aggregator jar.  See #4347

**Comment by pxLi** [comment]
fixed in #4355
